glue_vispy_viewers/common/toolbar.py

```diff
@@ -125,31 +172,163 @@ def on_mouse_release(self, event):
         z_att = self._vispy_widget.options.z_att
 
         # Get the visible datasets
-        visible_data = self.get_visible_data()
+        visible_data, visual = self.get_visible_data()
+        if len(visible_data) == 1:
+            layer = visible_data[0]
+        layer_data = np.array([layer[x_att], layer[y_att], layer[z_att]]).transpose()
+        print('layer data', layer_data, layer_data.shape)  #(1064, 3)
+        tr = visual.node_transform(self._vispy_widget.view)
+        print('tr', tr)
+
+        self._scatter = visual
+        self.scatter_data = layer_data
+
+        if event.button == 1 and self.mode is not None and self.mode is not 'point':
+            # self.facecolor[self.facecolor[:, 1] != 1.0] = self.white
+            data = tr.map(layer_data)[:, :2]
+
+            selection_path = path.Path(self.line_pos, closed=True)
+
+            mask = [selection_path.contains_points(data)]
+
+            self.selected = mask
+            self.mark_selected()
```

Comment:
I use this `mark_selected()` for highlighting to show that the mask is corrected extracted, it should be replaced with the `Subset` operation later.